server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstractionResolver.java

```diff
@@ -110,6 +110,8 @@ && isIndexVisible(
                 if (minus) {
                     finalIndices.remove(indexAbstraction);
                 } else if (indicesOptions.ignoreUnavailable() == false || isAuthorized.test(indexAbstraction)) {
+                    // only unauthorized names are considered "unavailable" at this stage
```

Comment:
I find these comments really hard to parse... is there any way to simplify either the code or the comments to make this more readable ? I think the biggest issue is the double (tripple?) negatives with a conflation of  terms... for  example "I do not want to ignore expressions that are not available"  is already difficult to parse and introducing treating unauthorized as unavailble reads "I want to treat unauthorized as unavailable which means that if I want to ignore expressions that not available ... I'm too lost to finish ".   

The question I want answered here is why do we care about unavailble at all here ?  this is the security code so it makes to test if authorized, but if the core code will remove unavailable as needed then why do i need to even look at the ignore flag here?

Reply:
The code comment that I added here is more confusing than helpful, apologies.

I changed the comment to:
```
// Unauthorized names are considered unavailable, so if `ignoreUnavailable` is `true` they should be silently
// discarded from the `finalIndices` list. Other "ways of unavailable" must be handled by the action
// handler, see: https://github.com/elastic/elasticsearch/issues/90215
```
Hope this makes it clearer.